components/tools/OmeroJava/test/integration/thumbnail/SkipThumbnailsPermissionsTest.java

```diff
@@ -0,0 +1,686 @@
+        //member regardless of their role cannot link to Pixels.
+        if (permissions.equalsIgnoreCase("rw----")) {
+            return;
+        }
+
+        EventContext user1 = newUserAndGroup(permissions);
+        loginUser(user1);
+
+        ImportConfig config = new ImportConfig();
+        config.doThumbnails.set(false); // skip thumbnails
+
+        // Create a fake file. A pyramid will be generated
+        Pixels pixels = importLargeFile(config);
+        final long pixelsId = pixels.getId().getValue();
+        RenderingEnginePrx re = factory.createRenderingEngine();
+        re.lookupPixels(pixelsId);
+        if (!re.lookupRenderingDef(pixelsId)) {
+            re.resetDefaultSettings(true);
+            re.lookupRenderingDef(pixelsId);
+        }
+        disconnect();
+        // Create new user in group and login as that user
+        EventContext user2 = newUserInGroup(user1, isGroupOwner);
+
+        // If user is an admin, add them to the system group
+        if (isAdmin) {
+            ExperimenterGroup systemGroup = new ExperimenterGroupI(iAdmin.getSecurityRoles().systemGroupId, false);
+            addUsers(systemGroup, Collections.singletonList(user2.userId), false);
+        }
+
+        // Login as user2
+        loginUser(user2);
+
+        // Generate rendering settings for user 2
+        re = factory.createRenderingEngine();
+        re.lookupPixels(pixelsId);
+        if (!re.lookupRenderingDef(pixelsId)) {
+            re.resetDefaultSettings(true);
+            re.lookupRenderingDef(pixelsId);
+        }
+        re.close();
+
+        // Load thumbnail as user 2 to create thumbnail on disk
+        ThumbnailStorePrx svc = factory.createThumbnailStore();
+        Utils.setThumbnailStoreToPixels(svc, pixelsId);
+        byte[] user2Thumbnail = Utils.getThumbnailWithoutDefault(svc);
+        Utils.checkSize(user2Thumbnail, Utils.DEFAULT_SIZE_X,
+                Utils.DEFAULT_SIZE_Y);
+        svc.close();
+
+        // Switch to user 1
+        disconnect();
+        loginUser(user1);
+
+        // Load and change to trigger rendering settings and thumbnail creation for user 1
+        svc = factory.createThumbnailStore();
+        Utils.setThumbnailStoreToPixels(svc, pixels.getId().getValue());
+
+        // Get rendering settings for pixels object as user 1
+        re = factory.createRenderingEngine();
+        re.lookupPixels(pixelsId);
+        if (!re.lookupRenderingDef(pixelsId)) {
+            re.resetDefaultSettings(true);
+            re.lookupRenderingDef(pixelsId);
+        }
+        re.load();
+        re.setActive(0, false);
+        re.saveCurrentSettings();
+        re.close();
+
+        // Get thumbnail for user 1
+        byte[] user1Thumbnail = Utils.getThumbnailWithoutDefault(svc);
+        Utils.checkSize(user1Thumbnail, 96, 96);
+        svc.close();
+
+        // Check that the thumbnails are different
+        Assert.assertFalse(Arrays.equals(user1Thumbnail, user2Thumbnail));
```

Comment:
Should be an array-aware `assertEquals`?

Reply:
What we need will be an ``AssertNotEquals`` but the method takes Object so I don't think it will give what we need

Reply:
Indeed you're correct: interesting! Never noticed there was more `assertEquals` than `assertNotEquals`.